src/components/App/App.jsx

```diff
@@ -1,20 +1,57 @@
 import React from 'react';
-import './App.styl';
+import styles from './App.styl';
 import Header from '../Header/Header';
 import MainContent from '../MainContent/MainContent';
 import Footer from '../Footer/Footer';
 import WithWrapper from '../WithWrapper/WithWrapper';
+import fetchMovies from '../../utils/fetchMovies';
+import normalizeMoviesData from '../../utils/normalizeMoviesData';
+import MainContextProvider from '../MainContextProvider/MainContextProvider';
+import ModalsWrapper from '../ModalsWrapper/ModalsWrapper';
+import { modalType } from '../Modal/Modal';
 
 const MainContentWithWrapper = WithWrapper(MainContent);
+class App extends React.Component {
+  state = {
+    moviesList: [],
+    isLoading: true,
+    currentModal: modalType.none,
+    isModalShown: false
+  }
 
-const App = () => {
-  return (
-    <div className="App">
-      <Header />
-      <MainContentWithWrapper />
-      <Footer />
-    </div>
-  );
-};
+  render() {
+    const {
+      isLoading,
+      moviesList,
+      currentModal,
+      isModalShown
+    } = this.state;
+
+    const openModal = modalType => this.setState({...this.state, isModalShown: true, currentModal: modalType});
+
+    const closeModal = () => this.setState({...this.state, isModalShown: false, currentModal: modalType.none});
```

Comment:
How do you look on moving these two methods outside `render` method so they won't be recreated for every render?

One more thing here to mention is that you don't need to copy all existing state if you need to change only small part of it. You can simply change only the part you need, f.e.:
```
openModal(modalType) {
   this.setState({ isModalShown: true, currentModal: modalType });
}
```

Reply:
Thanks, done